internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -202,6 +204,37 @@ func computeCostComponents(region, machineType string, purchaseOption string, in
 
 }
 
+func getSustainedUseDiscount(monthlyHours *float64, thresholds []float64, rates []float64, incrementalRates []float64) float64 {
+	// Assume 730 hours(max) if monthly_hrs is not set
+	if *monthlyHours == 0 {
+		*monthlyHours = 730.0
+	}
+	totalHoursInMonth := 730.0
+	totalDiscount := 0.0
+	remainingHours := *monthlyHours
+
+	for i, threshold := range thresholds {
```

Comment:
We should check that the lengths of thresholds, rates and incrementalRates all match to avoid any index errors.

Reply:
We should also check that the lengths are all greater than 0, and if not then return 0.0.